tests/components/nexia/test_config_flow.py

```diff
@@ -74,3 +74,44 @@ async def test_form_cannot_connect(hass):
 
     assert result2["type"] == "form"
     assert result2["errors"] == {"base": "cannot_connect"}
+
+
+async def test_form_import(hass):
+    """Test we get the form with import source."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    with patch(
+        "homeassistant.components.nexia.config_flow.NexiaHome.get_name",
+        return_value="myhouse",
+    ), patch(
+        "homeassistant.components.nexia.config_flow.NexiaHome.login",
+        side_effect=MagicMock(),
+    ), patch(
+        "homeassistant.components.nexia.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.nexia.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={CONF_USERNAME: "username", CONF_PASSWORD: "password"},
+        )
+
+    assert result["type"] == "create_entry"
+    assert result["title"] == "myhouse"
+    assert result["data"] == {
+        CONF_USERNAME: "username",
+        CONF_PASSWORD: "password",
+    }
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+    result2 = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_IMPORT},
+        data={CONF_USERNAME: "username", CONF_PASSWORD: "password"},
+    )
+
+    assert result2["type"] == "abort"
+    assert result2["reason"] == "already_configured"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5444210)